### ISSUE TITLE
geth 1.4.3 panics when calling RPC

### ISSUE THREAD

**Issue by slothbag** [ethereum/go-ethereum#2555]
Geth version: 1.4.3
OS & Version: Ubuntu Linux 14.04 64bit

RPC call is `eth_call`

```
github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest.func1(0xc823218ed0, 0x7f62ede0cb60, 0xc82000b810)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/server.go:153 +0xb7
panic(0xe9a280, 0xc820014080)
        /opt/google/go/src/runtime/panic.go:443 +0x4e9
encoding/json.(*decodeState).unmarshal.func1(0xc8206ed110)
        /opt/google/go/src/encoding/json/decode.go:154 +0x88
panic(0xe9a280, 0xc820014080)
        /opt/google/go/src/runtime/panic.go:443 +0x4e9
github.com/ethereum/go-ethereum/eth/filters.(*NewFilterArgs).UnmarshalJSON(0xc82327f180, 0xc820156481, 0xb3, 0xbf, 0x0, 0x0)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/eth/filters/api.go:387 +0x1676
encoding/json.(*decodeState).object(0xc8232831d0, 0xcd6d60, 0xc82040af40, 0x194)
        /opt/google/go/src/encoding/json/decode.go:560 +0x143
encoding/json.(*decodeState).value(0xc8232831d0, 0xcd6d60, 0xc82040af40, 0x194)
        /opt/google/go/src/encoding/json/decode.go:367 +0x3a1
encoding/json.(*decodeState).array(0xc8232831d0, 0xcbbdc0, 0xc82329e360, 0x197)
        /opt/google/go/src/encoding/json/decode.go:518 +0xa6b
encoding/json.(*decodeState).value(0xc8232831d0, 0xc8c540, 0xc82329e360, 0x16)
        /opt/google/go/src/encoding/json/decode.go:364 +0x3c1
encoding/json.(*decodeState).unmarshal(0xc8232831d0, 0xc8c540, 0xc82329e360, 0x0, 0x0)
        /opt/google/go/src/encoding/json/decode.go:168 +0x196
encoding/json.Unmarshal(0xc820156480, 0xb5, 0xc0, 0xc8c540, 0xc82329e360, 0x0, 0x0)
        /opt/google/go/src/encoding/json/decode.go:96 +0x12b
github.com/ethereum/go-ethereum/rpc.parsePositionalArguments(0xc820156480, 0xb5, 0xc0, 0xc8204a4820, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/json.go:273 +0x2b5
github.com/ethereum/go-ethereum/rpc.(*jsonCodec).ParseRequestArguments(0xc82000b810, 0xc8204a4820, 0x1, 0x1, 0xce2520, 0xc82329e340, 0x0, 0x0, 0x0, 0x0, ...)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/json.go:260 +0x19c
github.com/ethereum/go-ethereum/rpc.(*Server).readRequest(0xc823218ed0, 0x7f62ede0cb60, 0xc82000b810, 0x0, 0x0, 0x0, 0x7f62ede394b0, 0x0, 0x0)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/server.go:438 +0x156c
github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest(0xc823218ed0, 0x7f62ede0cb60, 0xc82000b810, 0xc820427901, 0x1, 0x0, 0x0)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/server.go:183 +0x52c
github.com/ethereum/go-ethereum/rpc.(*Server).ServeSingleRequest(0xc823218ed0, 0x7f62ede0cb60, 0xc82000b810, 0x1)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/server.go:234 +0x44
github.com/ethereum/go-ethereum/rpc.newJSONHTTPHandler.func1(0x7f62ede0ca50, 0xc823286340, 0xc82015e540)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/http.go:121 +0x6a4
net/http.HandlerFunc.ServeHTTP(0xc8204a4980, 0x7f62ede0ca50, 0xc823286340, 0xc82015e540)
        /opt/google/go/src/net/http/server.go:1618 +0x3a
github.com/rs/cors.(*Cors).Handler.func1(0x7f62ede0ca50, 0xc823286340, 0xc82015e540)
        /tmp/geth2/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rs/cors/cors.go:122 +0x17e
net/http.HandlerFunc.ServeHTTP(0xc82050f0c0, 0x7f62ede0ca50, 0xc823286340, 0xc82015e540)
        /opt/google/go/src/net/http/server.go:1618 +0x3a
net/http.serverHandler.ServeHTTP(0xc8203c5c00, 0x7f62ede0ca50, 0xc823286340, 0xc82015e540)
        /opt/google/go/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc820730480)
        /opt/google/go/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
        /opt/google/go/src/net/http/server.go:2137 +0x44e
```

**Comment by karalabe** [comment]
@bas-vk ?

**Comment by fjl** [comment]
I cannot reproduce this. Have tried `geth --rpc --rpcaddr "10.0.1.8"` and it does not crash.

@slothbag which command-line flags did you use?

**Comment by slothbag** [comment]
Turns out the daemon does start up successfully, but a remote client calling the RPC is causing the panic straight away.  I will update the title of issue and try find which call is the cause.

**Comment by slothbag** [comment]
I suspect it is related to this issue https://github.com/ethereum/go-ethereum/issues/2319

I'm calling the following repeatedly.. and eventually geth gives the panic message documented earlier in this issue.  But it appears geth continues to run.
```
{"id":1,"jsonrpc":"2.0","method":"eth_call","params":[{"from":null,"to":"0x73cc8baf770e17af57c1519abac8708e75b7d6e9","gas":null,"gasPrice":null,"value":null,"data":"0x9d76ea58"},"latest"]}
```

The response..
```
Invalid address length, expected 40 got 4 bytes
```

**Comment by bas-vk** [comment]
From the stack trace it looks like `eth_newFilter` is causing this and not `eth_call`.
Could you post the request that creates the filter?

**Comment by slothbag** [comment]
I can't tell if this is the culprit cause i'm getting the rpc data from wireshark.. but here is the eth_newFilter call

```
[truncated] {"id":1,"jsonrpc":"2.0","method":"eth_newFilter","params":[{"fromBlock":"0xcd4a2","toBlock":"latest","address":["0x70c87d191324e6712a591f304b4eedef6ad9bb9d"],"topics":[["cc26ef7e6d83605f4a7e3e740abde741630f511bc3a769e7934484fbd
```

**Comment by bas-vk** [comment]
Thank you, I'm able to reproduce it and wil create a fix.

**Comment by slothbag** [comment]
Do you think the `Invalid address length, expected 40 got 4 bytes` is due to the same bug? or should I chase that up in another issue?

**Comment by slothbag** [comment]
I just tried geth 1.4.4, same panic occuring with that new_filter call.  geth keeps running.

If no one reads this, i'll open a new issue in 24 hours.

**Comment by bas-vk** [comment]
@slothbag, this is solved in the referenced PR that still needs to be merged.

**Comment by slothbag** [comment]
Oh right, in the release notes for 1.4.4 it says `Fixed: crash when assigning --rpcaddr #2555 (#2563)`

Which points to this issue.  So I assumed it was merged already. Sorry
